sklearn/metrics/_ranking.py

```diff
@@ -532,7 +532,10 @@ def _binary_clf_curve(y_true, y_score, pos_label=None, sample_weight=None):
              np.array_equal(classes, [0]) or
              np.array_equal(classes, [-1]) or
              np.array_equal(classes, [1]))):
-        raise ValueError("Data is not binary and pos_label is not specified")
+        raise ValueError("y_true takes value in {classes} and pos_label is "
```

Comment:
Do we need something like the following?

```py
if (pos_label is None and _determine_key_type(classes) == 'str'
    ...
```

Reply:
Thanks indeed, I updated my PR based on your suggestion and I could get rid of the remaining numpy `FutureWarning` in `array_equal`.